controllers/bindings/client_factory.go

```diff
@@ -186,6 +189,12 @@ func (cf *CachingClientFactory) GetClient(ctx context.Context, currentNamespace
 			ttl = cf.MaxClientCacheTTL
 		}
 
+		_, err = cl.(client.Client).RESTMapper().ResourcesFor(schema.GroupVersionResource{Version: "v1", Resource: "services"})
+		if err != nil {
+			log.FromContext(ctx).Info("Not able to list available resources with target k8s client", "host", cfg.Host, "error", err.Error())
```

Comment:
Could we add a line to the documentation here https://github.com/redhat-appstudio/remote-secret/blob/main/docs/USER.md#another-cluster that we are fetching the services just to check that the credentials are valid and we do not use the result in any other way?

Reply:
I think we are using the credentials later in the flow. This is just to catch the invalid connection sooner in the flow so that we can diagnose it better.

Reply:
> This is just to catch the invalid connection sooner
Yes, my thinking is that somebody who gave us cluster credentials might be looking at the requests and ask himself why we are fetching services. But maybe I am paranoid, and it is not a problem :D